pyproject.toml

```diff
@@ -25,32 +26,33 @@ classifiers = [
     "Operating System :: Unix",
     "Operating System :: MacOS",
 ]
-requires-python = ">=3.9"
-dependencies = [
-    "aws-cdk-lib==2.48.0",
-    "aws-cdk.aws-lambda-python-alpha",
-    "constructs>=10.0.0,<11.0.0",
-    "boto3",
-    "opensearch-py",
-    "requests",
-    "python-jose"
-]
 
-[project.optional-dependencies]
-dev = [
-    "pytest==6.2.5",
-    "moto"
-]
+[tool.poetry.dependencies]
+python = ">=3.9,<4"
+# Required for aws-cdk.aws-lambda-python-alpha
+aws-cdk-lib = ">=2.66.0,<3.0.0"
+#aws-cdk-lib = "2.48.0"
+boto3 = "^1.26.78"
+"aws-cdk.aws-lambda-python-alpha" = "^2.66.0a0"
+constructs = ">=10.0.0,<11.0.0"
+black = "^23.1.0"
+
+[tool.poetry.group.tests.dependencies]
+# Installed by default, can be installed without these dependencies using `poetry install --without tests`
+pytest = "==6.2.5"
+moto = "^4.1.3"
+hypothesis = "^6.68.2"
+
+[tool.poetry.group.cdk]
+# Not installed by default, can be installed using `poetry install --with cdk`
+optional = true
+
+[tool.poetry.group.cdk.dependencies]
```

Comment:
If you can export this to a `poetry.lock` file in the lambda area that would be great! It should be placed in `sds_data_manager/lambda_code/poetry.lock` where there is a current requirements.txt file I think.

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-lambda-python-alpha-readme.html#packaging

Reply:
Ok, will investigate and add that! Good idea!

Reply:
Looks like poetry will only export to one type of `poetry.lock` file. Then, the lambda constructor will convert that to a `requirements.txt`. So, our options are either to copy the poetry.lock file into the lambda subfolder, or export the poetry project to a requirements.txt file to replace the one that's currently in there. Either way, I think that should be set up as a precommit action ([see here)](https://python-poetry.org/docs/cli/#export) and all dependencies should still be managed through this poetry file.

Reply:
The thing we don't want is all the CDK dependencies going into the lambda_dev requirements (would include a bunch of AWS package bloat), we only want the lambda runtime requirements over there. So this might be a bit of work to figure out how to thread through only that section's dependencies and not the "project"'s dependencies.

Reply:
Yeah, in that case, we'll want to generate a requirements.txt file (which can select different groups to include or exclude) since if AWS generates it we won't have control over what gets installed.

Reply:
> since if AWS generates it we won't have control over what gets installed.

Right now, it is just hand-made, so we have to manually keep track of the dependencies. I suppose you could also handmake a poetry.lock file?

Reply:
It won't be hand made. You can generate a requirements.txt file from the poetry dependencies (using the command `poetry export --without tests --with lambda_dev`) which will only contain the groups that you want included. 

On the other hand, the poetry.lock file includes everything, without any way to change it, but you control what's installed using the `poetry install` command.

 Since AWS will basically take the poetry.lock file and run `poetry export` without specifying groups, it will install everything out of poetry.lock. So instead, we should set up a pre-commit hook which will generate a requirements.txt file using only the specific groups we want to include. 

That generated file should NOT be changed by hand, all the dependencies should still be managed through pyproject.toml.

Reply:
Alright @greglucas, I added the precommit hook and the requirements.txt. Can you take a look? Thanks!